### PR TITLE
Add Rethrow and Fix Exn

### PR DESCRIPTION
**Pull request by matjin**
This PR introduces:

1. Translation of rethrow
2. Handles a rare exception that can occur if the top snapshot instruction is null.

### CODE REVIEW

File: Cilsil/Services/CfgParserService.cs [hunk 1]
```diff
@@ -199,6 +199,14 @@ private void ComputeMethodCfg(MethodDefinition method)
                     programState.PushInstruction(methodBody.Instructions.First(), initNode);
                     do
                     {
+                        if (programState.IsTopSnapshotInstructionNull())
```
Comment (xi-liu-ds) [inline Cilsil/Services/CfgParserService.cs:202]: Considering at line 199 an instruction was pushed into `InstructionStack` and the while loop is condition on `InstructionStack.count > 0`, why is this case possible?

File: Cilsil/Cil/Parsers/NewobjParser.cs [hunk 2]
```diff
@@ -51,13 +51,13 @@ internal class NewobjParser : InstructionParser
...
                     // Represents constructor call; we discard the return var as it's not needed.
                     CreateMethodCall(state,
-                                     false,
-                                     constructorMethod,
-                                     out _,
-                                     out _,
-                                     out _,
-                                     out var constructorCall,
-                                     isConstructorCall: true);
+                                        false,
```
Comment (xi-liu-ds) [inline Cilsil/Cil/Parsers/NewobjParser.cs:54]: Redundant indent

File: Cilsil/Services/CfgParserService.cs (hunk 1 above)
Comment (matjin) [reply Cilsil/Services/CfgParserService.cs:202]: I haven't looked too deeply into the root causes, but this is a very rare case that was calling a nullreference exception in the translation, so I handle it here (and log it)

File: Cilsil/Cil/Parsers/NewobjParser.cs (hunk 2 above)
Comment (matjin) [reply Cilsil/Cil/Parsers/NewobjParser.cs]: done

**Review by xi-liu-ds — APPROVED** [review]

**Pull request merged**